bin/aurelia.js

```diff
@@ -1,38 +1,134 @@
-  configName: 'Aureliafile',
-  extensions: require('interpret').jsVariants,
-  // ^ automatically attempt to require module for any javascript variant
-  // supported by interpret.  e.g. coffee-script / livescript, etc
+#!/usr/bin/env node --harmony
+
+'use strict';
+
+var argv       = require('minimist')(process.argv.slice(2));
+var chalk      = require('chalk');
+var pjson      = require('../package.json');
+
+var AureliaCLI = require('../AureliaCLI.js');
+
+var cli = new AureliaCLI({
+  name       : 'aurelia-cli',
+  configName : 'Aureliafile',
+  argv       : argv,
+  cwd        : process.cwd(),
+  verbose    : argv.verbose,
+  launchFile : 'lib/program.js',
+  extensions : require('interpret').jsVariants,
+  //   // ^ automatically attempt to require module for any javascript variant
+  //   // supported by interpret.  e.g. coffee-script / livescript, etc
   v8flags: ['--harmony'] // to support all flags: require('v8flags')
-  // ^ respawn node with any flag listed here
+  //   // ^ respawn node with any flag listed here
 });
 
-AureliaCLI
-  .on('require', function(name, module) {
-    console.log('Loading:', name);
-  })
-  .on('requireFail', function(name, err) {
-    console.log('Unable to load:', name, err);
-  })
-  .on('respawn', function(flags, child) {
-    console.log('Detected node flags:', flags);
-    console.log('Respawned to PID:', child.pid);
-  });
-
-var cli = require('../lib/cli');
-
-cli.configure({
-  argv: argv,
-  cwd: argv.cwd,
-  configPath: argv.aureliafile,
-  require: argv.require,
-  completion: argv.completion,
-  verbose: argv.verbose,
-  Lyftoff: AureliaCLI,
-  launchFile: 'lib/program.js'
+process.title = cli.processTitle;
+process.AURELIA = cli;
+
+var logger = cli.logger;
+
+// EVENT init
+// This event will be emitted regardless of the users environment.
+// If the user is missing the config file
+// If the user is missing the locally installed Aurelia-cli
+// This function will always be invoked.
+cli.initialize(function (program) {
+
+  if (argv.verbose) {
+    logger.log('LIFTOFF SETTINGS:', this.env);
+    logger.log('CLI OPTIONS:', this.env.argv);
+    logger.log('CWD:', this.env.cwd);
+    // log('LOCAL MODULES PRELOADED:', env.require);
+    // log('SEARCHING FOR:', env.configNameRegex);
+    logger.log('FOUND CONFIG AT:', this.env.configPath);
+    logger.log('CONFIG NAME:', this.env.configName);
+    logger.log('YOUR LOCAL MODULE IS LOCATED:', this.env.modulePath);
+    logger.log('LOCAL PACKAGE.JSON:', this.env.modulePath);
+    logger.log('CLI PACKAGE.JSON', require('../package'));
+  }
+
+  program.version(pjson.version)
+    .on('--help', function () {
+      console.log(['\n'
+        , '    ' + chalk.bgMagenta(' ') + chalk.bgRed('  ') + chalk.bgMagenta(' ') + '\n'
+        , '  ' + chalk.bgRed.black(' aurelia ') + '\n'
+        , '    ' + chalk.bgMagenta(' ') + chalk.bgRed('  ') + chalk.bgMagenta(' ') + '\n'
+        ]).join('');
+    });
+
+  program.command('new <type>')
```

Comment:
All these command registration should be in a separate file. We will have to amend this every time we add a new command or change the option of a command. It makes more sense if we separate it from the cli initialization.